src/ert/storage/local_ensemble.py

```diff
@@ -237,23 +245,32 @@ def load_ext_param(self, key: str, realization: int) -> Any:
             data = json.load(f)
         return data
 
-    def has_surface(self, key: str, realization: int) -> bool:
-        input_path = self.mount_point / f"realization-{realization}"
-        return (input_path / f"{key}.irap").exists()
-
-    def load_surface_file(self, key: str, realization: int) -> RegularSurface:
-        input_path = self.mount_point / f"realization-{realization}" / f"{key}.irap"
-        if not input_path.exists():
-            raise KeyError(f"No parameter: {key} in storage")
-        surf = xtgeo.surface_from_file(input_path, fformat="irap_ascii")
-        return surf
+    def _load_dataset(
+        self,
+        group: str,
+        realizations: Union[int, Sequence[int]],
+    ) -> xr.Dataset:
+        def mkpath(index: int) -> Path:
+            path = self.mount_point / f"realization-{index}" / group
+            if not path.is_file():
+                raise KeyError(f"No parameter: {group} in storage")
+            return path
+
+        if isinstance(realizations, int):
+            path = mkpath(realizations)
+            return xr.open_dataset(path, engine="scipy")
+
+        return xr.open_mfdataset(
```

Comment:
Did you check the performance of `mfdataset` compared with `open_dataset`?

Reply:
No. I assume it will be bad. I will change it to do the same thing as you do in your PR.

Reply:
You are more explicit in the call though, so might be that there is little performance gain. So maybe this is the way.